lib/elixir/lib/exception.ex

```diff
@@ -765,6 +765,24 @@ defmodule UnicodeConversionError do
   end
 end
 
+defmodule StartLinkNameError do
+  defexception [arg: nil]
+
+  def message(exception) do
+    """
+    invalid argument type given for `name`.
+
+    Valid types are:
+      * nil
+      * atom
+      * tuple
```

Comment:
We just have `is_tuple/1` because we are being a bit lazy. If a tuple it must be:
```elixir
{:global, term} | {:via, module, term}
```